### PR TITLE
DB/Quest: No Mere Dream

### LINKED ISSUE #11252: {3.3.5a} [Quest] No Mere Dream
Related Issue Report: #8689

Quest: [No Mere Dream](http://www.wowhead.com/quest=10965)
QuestGiver: [Clintar Dreamwalker](http://www.wowhead.com/npc=22834)
Related: [Clintar Dreamwalker's Spirit](http://www.wowhead.com/npc=22916)

Problem: From what I've seen, you can complete the quest only once per session.
In every next case Clintar's spirit becomes either
* Invisible, making it impossible to track unless turn GM status
* Invisible, making it impossible to track even if you DO have GM status
* Turns into a ghost [Like this but green in color](http://wow.zamimg.com/uploads/screenshots/normal/89933.jpg)
* Turns into a cat (critter cat model).

I don't know what model he should have but I think it should be one of Druid's Cat forms.
In majority of these cases, quest does not get rewarded for completion when you help him.

Revision: 055af93145a636913e2592993aa87e828ee00a66
TDB 52 + Repo updates

<bountysource-plugin>
---
Want to back this issue? **[Post a bounty on it!](https://www.bountysource.com/issues/1412737-3-3-5a-quest-no-mere-dream?utm_campaign=plugin&utm_content=tracker%2F1310&utm_medium=issues&utm_source=github)** We accept bounties via [Bountysource](https://www.bountysource.com/?utm_campaign=plugin&utm_content=tracker%2F1310&utm_medium=issues&utm_source=github).
</bountysource-plugin>

### PR DESCRIPTION
**Pull request by Lumber-Jack-2**
- by @Vincent-Michael + @Killyana 
  closes #11252

### CODE REVIEW

File: src/server/scripts/Kalimdor/zone_moonglade.cpp [hunk 1]
```diff
@@ -300,6 +300,9 @@ class npc_clintar_spirit : public CreatureScript
         {
             Initialize();
             PlayerGUID.Clear();
+            me->SetDisplayId(me->GetCreatureTemplate()->Modelid1);
+            me->RemoveFlag(UNIT_FIELD_FLAGS, UNIT_FLAG_NOT_SELECTABLE);
+            me->RemoveFlag(UNIT_FIELD_FLAGS, CREATURE_FLAG_EXTRA_TRIGGER);
```
Comment (joschiwald) [inline src/server/scripts/Kalimdor/zone_moonglade.cpp]: what????
remove the flag in db if you don't need it

Comment (Rushor) [inline src/server/scripts/Kalimdor/zone_moonglade.cpp]: well i don't know if sniffs confirmed these flags or not

Comment (Killyana) [inline src/server/scripts/Kalimdor/zone_moonglade.cpp]: Yes the npc have this flags on sniffs, they are removed when the script starts, they are used to make it invisible to the player if the npc is not active.

Comment (joschiwald) [inline src/server/scripts/Kalimdor/zone_moonglade.cpp]: blizz does not send triggerflags in UnitFields

**Pull request merged**